packages/@lwc/module-resolver/src/index.ts

```diff
@@ -6,154 +6,109 @@
  */
 /* eslint-env node */
 
-import glob from 'fast-glob';
 import path from 'path';
 import fs from 'fs';
-import nodeModulePaths from './node-modules-paths';
-
-const DEFAULT_IGNORE = ['**/node_modules/**', '**/__tests__/**'];
-const PACKAGE_PATTERN = ['*/*/package.json', '*/package.json', 'package.json'];
-const MODULE_ENTRY_PATTERN = `**/*.([jt]s|html|css)`;
-const LWC_CONFIG_FILE = '.lwcrc';
+import {
+    getModuleEntry,
+    normalizeConfig,
+    loadConfig,
+    mergeModules,
+    LWC_CONFIG_FILE,
+} from './utils';
 
 export interface RegistryEntry {
     entry: string;
-    moduleSpecifier: string;
-    moduleName: string;
-    moduleNamespace: string;
+    specifier: string;
 }
 
-export interface ModuleResolverConfig {
-    moduleDirectories: string[];
-    rootDir: string;
-    modulePaths: string[];
-    ignorePatterns?: string[];
+export interface ModuleRegistryMap {
+    [key: string]: RegistryEntry;
 }
 
-interface FlatEntry {
-    path: string;
+export interface ModuleResolverConfig {
+    rootDir: string;
+    modules: ModuleId[];
 }
 
-function createRegistryEntry(entry, moduleSpecifier, moduleName, moduleNamespace): RegistryEntry {
-    return { entry, moduleSpecifier, moduleName, moduleNamespace };
+export type ModuleId = string | [string, string];
+export interface LwcConfig {
+    modules: ModuleId[];
 }
 
-function loadLwcConfig(modulePath) {
-    const packageJsonPath = path.join(modulePath, 'package.json');
-    const lwcConfigPath = path.join(modulePath, LWC_CONFIG_FILE);
-    let config;
-    try {
-        config = JSON.parse(fs.readFileSync(lwcConfigPath, 'utf8'));
-    } catch (ignore) {
-        // ignore
-    }
-    if (!config) {
-        try {
-            config = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8')).lwc;
-        } catch (ignore) {
-            // ignore
+export function resolveModulesFromDir(modulesDir: string): RegistryEntry[] {
+    const namespaces = fs.readdirSync(modulesDir);
+    const resolvedModules: RegistryEntry[] = [];
+    namespaces.forEach(ns => {
+        // TODO: A better isDir() bellow
```

Comment:
should this be addressed this time?

Reply:
I was in the plane, I just couldn't figure out the APIs xD